.travis.yml

```diff
@@ -25,8 +38,39 @@ env:
     - TARGET_PLATFORM=ubuntu/bionic
     - TARGET_PLATFORM=ubuntu/xenial
 before_install:
-  - export PACKAGE_ENCRYPTION_KEY=verysecret
-  - git clone -b v0.7.14 --depth 1 https://github.com/citusdata/tools.git
+  - git clone -b simplify-branching --depth 1 https://github.com/citusdata/tools.git
   - sudo make -C tools install
+  - |
+    if [ -n "${MICROSOFT_PACKAGE:-}" ]; then
+    export PACKAGE_ENCRYPTION_KEY=verysecret
+    else
+    export PACKAGING_SECRET_KEY="$(openssl aes-256-cbc -K $encrypted_a39e3f1ee8ba_key -iv $encrypted_a39e3f1ee8ba_iv -in signing_key.asc.enc -d | base64)"
+    fi
 install: true
-script: "export PACKAGE_ENCRYPTION_KEY=verysecret && citus_package -p ${TARGET_PLATFORM} 'local' 'release'"
+script: |
```

Comment:
As far as I understand, you wanted to change the CI behaviour here depending on MS and non-MS packages.

MS package: we will try to build the packages and we will never attempt to push them to PackageCloud. It will be sufficient to create the package

Non-MS packages: similar to what we already have, we will create the packages, and upload them to packagecloud once the PR is merged.

Is my understanding correct?

Reply:
Yes that is correct. One thing to note is that teh MS packages also test that the encryption is working.